### DISCUSSION TITLE
How to enable opentelemetry in Milvus

### DISCUSSION THREAD

**Discussion by morningspace** [milvus-io/milvus#30044] [Q&A and General discussion]
By reading the release history, it looks Milvus has supported OpenTelemetry since 2.3.0, but I was not able to find out how to enable it, e.g.: to enable and display tracing or metrics collecting. Did I miss something?

**Comment by LoveEachDay** [comment]
To enable tracing for milvus, you'd add the following config in your `milvus.yaml` :
```
trace:
  # trace exporter type, default is stdout,
  # optional values: ['stdout', 'jaeger', 'otlp']
  exporter: stdout
  # fraction of traceID based sampler,
  # optional values: [0, 1]
  # Fractions >= 1 will always sample. Fractions < 0 are treated as zero.
  sampleFraction: 0
  otlp:
    endpoint: # "127.0.0.1:4317"
    secure: true
  jaeger:
    url: # "http://127.0.0.1:14268/api/traces"
    # when exporter is jaeger should set the jaeger's URL
```
Here are some tips:
1. We only use OpenTelemetry to export tracing instead of metrics
2. You'd first choose an `exporter` type
3. By default we don't send any trace unless you change the `sampleFraction` to a number of (0,1]
4. You need to deploy a service to receive the exported trace.

**Comment by morningspace** [comment]
Thanks @LoveEachDay ! I will try that with my local milvus instance and get back to you if I have any other question. I wonder why this is not seen in the current docs, or did I miss it?

**Comment by morningspace** [comment]
Just to confirm that I need to switch to 2.3.4, so far the latest release, in order to support insecure, right? Otherwise I will get an error like this:

```console
2024/01/17 10:35:06 traces export: max retry time elapsed: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: tls: first record does not look like a TLS handshake"
```
I was using 2.3.3 previously.
